File: apps/dashboard/src/types/investigation.ts
export enum CaseStatus {
  OPEN = "OPEN",
  TRIAGING = "TRIAGING",
  INVESTIGATING = "INVESTIGATING",
  EVIDENCE_REVIEW = "EVIDENCE_REVIEW",
  DECIDED = "DECIDED",
  CLOSED = "CLOSED",
  ESCALATED = "ESCALATED",
}

export enum CasePriority {
  LOW = "LOW",
  MEDIUM = "MEDIUM",
  HIGH = "HIGH",
  CRITICAL = "CRITICAL",
}

export interface Finding {
  finding_id: string;
  agent_name: string;
  finding_type: string;
  description: string;
  confidence: number;
  evidence_refs: string[];
  timestamp: string;
}

export interface Evidence {
  evidence_id: string;
  type: string;
  source: string;
  content: string;
  metadata: Record<string, unknown>;
  timestamp: string;
}

export interface TimelineEvent {
  event_id: string;
  timestamp: string;
  event_type: string;
  description: string;
  agent_name: string;
  data?: Record<string, unknown>;
}

export interface InvestigationCase {
  case_id: string;
  transaction_ids: string[];
  entity_ids: string[];
  status: CaseStatus;
  priority: CasePriority;
  risk_score: number;
  assigned_to?: string;
  findings: Finding[];
  evidence: Evidence[];
  timeline: TimelineEvent[];
  created_at: string;
  updated_at: string;
  closed_at?: string;
  verdict?: string;
  sar_generated: boolean;
}

File: core/schemas/investigation.py
from enum import Enum
from typing import List, Dict, Any, Optional
from datetime import datetime
from pydantic import Field
from uuid import UUID
from .base import BaseSchema


class CaseStatus(str, Enum):
    OPEN = "OPEN"
    TRIAGING = "TRIAGING"
    INVESTIGATING = "INVESTIGATING"
    EVIDENCE_REVIEW = "EVIDENCE_REVIEW"
    DECIDED = "DECIDED"
    CLOSED = "CLOSED"
    ESCALATED = "ESCALATED"


class CasePriority(str, Enum):
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    CRITICAL = "CRITICAL"


class Finding(BaseSchema):
    finding_id: str
    agent_name: str
    finding_type: str
    description: str
    confidence: float
    evidence_refs: List[str]
    timestamp: datetime = Field(default_factory=datetime.utcnow)


class Evidence(BaseSchema):
    evidence_id: str
    type: str
    source: str
    content: Any
    metadata: Dict[str, Any]
    timestamp: datetime = Field(default_factory=datetime.utcnow)


class TimelineEvent(BaseSchema):
    event_id: str
    timestamp: datetime = Field(default_factory=datetime.utcnow)
    event_type: str
    description: str
    agent_name: str
    data: Dict[str, Any]


class InvestigationCase(BaseSchema):
    case_id: str
    transaction_ids: List[str]
    entity_ids: List[str]
    status: CaseStatus
    priority: CasePriority
    risk_score: float
    assigned_to: Optional[str] = None
    findings: List[Finding] = Field(default_factory=list)
    evidence: List[Evidence] = Field(default_factory=list)
    timeline: List[TimelineEvent] = Field(default_factory=list)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    closed_at: Optional[datetime] = None
    verdict: Optional[str] = None
    sar_generated: bool = False
